tic_tac_toe/game.py

```diff
@@ -87,18 +126,68 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
-
+    #check if player turn, check if valid move, check if empty position
+    
+    if get_winner(game):
+        raise InvalidMovement("Game is over.")
+        
+    if _board_is_full(game['board']):
+        raise InvalidMovement("Game is over.")
+     
+    if get_next_turn(game) != player:
+        raise InvalidMovement('"{}" moves next'.format(get_next_turn(game)))
+        
+    if not _position_is_valid(position):
+        raise InvalidMovement("Position out of range.")
+        
+    if not _position_is_empty_in_board(position, game['board']):
+        raise InvalidMovement("Position already taken.")
+   
+    game['board'][position[0]][position[1]] = player
+    
+    if player == 'X':
+        game['next_turn'] = 'O'
+    else:
+        game['next_turn'] = 'X'
+            
+    if _check_winning_combinations(game['board'], player):
+        game['winner'] = player
+        raise GameOver('"{}" wins!'.format(player))
+        
+    elif _board_is_full(game['board']):
+        raise GameOver('Game is tied!')
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+    '''expected = """
+    {0}  |  {1}  |  {2}
+    --------------
+    {3}  |  {4}  |  {5}
+    --------------
+    {6}  |  {7}  |  {8}
+    """.format(game['board'][0][0],game['board'][0][1],game['board'][0][2],
+                game['board'][1][0],game['board'][1][1],game['board'][1][2],
+                game['board'][2][0],game['board'][2][1],game['board'][2][2])
+                
+    return expected'''
+                
+                
+    board_template = """
+{0}  |  {1}  |  {2}
+--------------
+{3}  |  {4}  |  {5}
+--------------
+{6}  |  {7}  |  {8}
+""" 
+
```

Comment:
I like this board_template much better too. Again, much more visual, so easier to maintain.